File: backend/GeoQuiz_backend/GeoQuiz_backend/Application/DTOs/KingOfTheHill/RoundFinishedData.cs
using GeoQuiz_backend.Domain.Entities;

namespace GeoQuiz_backend.Application.DTOs.KingOfTheHill
{
    public class RoundFinishedData
    {
        public int RoundNumber { get; set; }
        public RoundType RoundType { get; set; }
        public List<Guid> EliminatedPlayerIds { get; set; } = new(); 
        public List<PlayerRoundResult> Results { get; set; } = new(); 
        public int RemainingPlayers { get; set; }
        public bool IsMatchFinished { get; set; }
    }

    public class PlayerRoundResult
    {
        public Guid PlayerId { get; set; }
        public bool HasAnswered { get; set; }
        public bool IsCorrect { get; set; }
        public int TimeSpentMs { get; set; }
        public int ScoreGained { get; set; }
    }
}

File: frontend/app/src/main/java/com/example/geoquiz_frontend/data/remote/dtos/koth/RoundFinishedData.java
package com.example.geoquiz_frontend.data.remote.dtos.koth;

import com.example.geoquiz_frontend.domain.enums.RoundType;

import java.util.List;

public class RoundFinishedData {
    private int roundNumber;
    private int roundType;
    private int correctOptionIndex;
    private List<String> eliminatedPlayerIds;
    private List<PlayerRoundResult> results;
    private int remainingPlayers;
    private boolean isMatchFinished;

    public int getRoundNumber() { return roundNumber; }
    public int getRoundTypeValue() { return roundType; }
    public RoundType getRoundType() { return RoundType.fromValue(roundType); }
    public int getCorrectOptionIndex() { return correctOptionIndex; }
    public List<String> getEliminatedPlayerIds() { return eliminatedPlayerIds; }
    public List<PlayerRoundResult> getResults() { return results; }
    public int getRemainingPlayers() { return remainingPlayers; }
    public boolean isMatchFinished() { return isMatchFinished; }

    public boolean isEliminated(String userId) {
        return eliminatedPlayerIds != null && eliminatedPlayerIds.contains(userId);
    }
}
